### PR TITLE
[WIP] Add conventions

### PR DESCRIPTION
**Pull request by unascribed**
Just some general conventions for mods, intended to be fairly generic.

I'm basing this on my own experiences and common 'de-facto' practices I've seen in actual mods.

Feedback welcome.

- [x] Introduction
- [x] Mod identity (name/modid/version)
- [x] Package and class naming
- [ ] Ore dictionary names
- [ ] Block/item naming
- [ ] Other?

### CODE REVIEW

**Comment by unascribed** [comment]
Any feedback, or should I just continue?

File: conventions/identity.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+A mod filename should never contain any of the characters that are illegal on most popular filesystems; these are:
+
+ - `#` (pound/hash/number sign)
+ - `<` (left angle bracket/less than)
+ - `>` (right angle bracket/greater than)
+ - `$` (dollar sign)
+ - `+` (plus sign)
+ - `%` (percent sign)
+ - `!` (exclamation point)
+ - <code>`</code> (backtick/grave)
```
Comment (tterrag1098) [inline conventions/identity.md]: No need for this, \\\` should work fine.

**Comment by tterrag1098** [comment]
#14 Duplicates some of your work, could you merge what you are doing with the changes made in that PR?

**Comment by unascribed** [comment]
I was participating in the creation of #14, when I got back to working on
this PR I was going to merge it. I've already merged it with my fork,

I also tried a variety of ways of escaping the backtick, including your
suggestion. It didn't work, so I wound up with what you see.

File: conventions/identity.md (hunk 1 above)
Comment (unascribed) [inline conventions/identity.md]: I tried a variety of ways of escaping the backtick, including your suggestion, and this is the only one that worked.

**Comment by tterrag1098** [comment]
Oh alright, good to hear.

File: conventions/identity.md (hunk 1 above)
Comment (tterrag1098) [inline conventions/identity.md]: Really? Ok. Let me mess around with it.

Comment (tterrag1098) [inline conventions/identity.md]: http://mcforge-tterrag.readthedocs.org/en/testing/conventions/identity/

https://github.com/tterrag1098/Documentation/commit/1a86747b3a6b6c625fc478232cadf537ffa23324#diff-0214770e7e042b22c304a056340db8b9R26

That seems to work. It's much cleaner, though I'm still not sure exactly *why* it works :P

**Comment by tterrag1098** [comment]
I finished reading all of it and it seems good, please continue.

Do note that Ore Dictionary "conventions" are not defined by forge, so that is unexplored territory. There are commonly used conventions yes, but no official ones.

File: conventions/identity.md (hunk 1 above)
Comment (tterrag1098) [inline conventions/identity.md]: Whoops I just nuked those links. The code was "\`\`\\\`\`"

**Comment by tterrag1098** [comment]
I updated the mkdocs.yml syntax to the latest version's syntax, so you'll have to adapt to that. Sorry.

**Comment by unascribed** [comment]
I know there's no real conventions for a lot of this stuff, which is why I'm trying to define some.

Also, yay new syntax! My local ignored mkdocs.yml has the new syntax as it wasn't working otherwise.

**Comment by tterrag1098** [comment]
Progress update?

**Comment by unascribed** [comment]
I've been working a lot on personal projects and work lately and haven't been doing much with Forge.
I'll try to pick this back up and get it at least partially finished.

**Comment by unascribed** [comment]
For the record, I'm unlikely to continue this PR at this point unless I randomly get motivation to work on it again. I've never been particularly good at writing and am having trouble explaining conventions in abstract enough terms to be understandable, while still being well enough defined to be useful. I'll close this PR, if someone wants to pick it back up and finish it, by all means. I'll reopen it if I stumble upon aforementioned magical motivation.

**Comment by bonii-xx** [comment]
I suppose it's OK if your work is getting salvaged though?

**Comment by unascribed** [comment]
@bonii-xx 
> if someone wants to pick it back up and finish it, by all means

Of course. :P

**Pull request closed without merging**